clouddriver-core/src/main/java/com/netflix/spinnaker/clouddriver/security/AccountDefinitionSource.java

```diff
@@ -0,0 +1,71 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.security;
+
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.netflix.spinnaker.credentials.definition.CredentialsDefinition;
+import com.netflix.spinnaker.credentials.definition.CredentialsDefinitionSource;
+import com.netflix.spinnaker.kork.annotations.NonnullByDefault;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Provides a full list of CredentialsDefinition account instances for a given credentials type.
+ * Given an {@link AccountDefinitionRepository} bean and an optional list of {@code
+ * CredentialsDefinitionSource<T>} beans for a given account type {@code T}, this class combines the
+ * lists from all the given credentials definition sources. When no {@code
+ * CredentialsDefinitionSource<T>} beans are available for a given account type, then a default
+ * source should be specified to wrap any existing Spring configuration beans that provide the same.
+ *
+ * @param <T> account credentials definition type
+ */
+@NonnullByDefault
+public class AccountDefinitionSource<T extends CredentialsDefinition>
+    implements CredentialsDefinitionSource<T> {
+
+  private final List<CredentialsDefinitionSource<T>> sources;
+
+  /**
+   * Constructs an account-based {@code CredentialsDefinitionSource<T>} using the provided
+   * repository, account type, and additional sources for accounts of the same type.
+   *
+   * @param repository the backing repository for managing account definitions at runtime
+   * @param type the account type supported by this source (must be annotated with {@link
+   *     JsonTypeName})
+   * @param additionalSources the list of other credential definition sources to list accounts from
+   */
+  public AccountDefinitionSource(
+      AccountDefinitionRepository repository,
+      Class<T> type,
+      List<CredentialsDefinitionSource<T>> additionalSources) {
+    var typeName = AccountDefinitionMapper.getJsonTypeName(type);
+    List<CredentialsDefinitionSource<T>> sources = new ArrayList<>(additionalSources.size() + 1);
+    sources.add(
+        () ->
+            repository.listByType(typeName).stream().map(type::cast).collect(Collectors.toList()));
+    sources.addAll(additionalSources);
+    this.sources = List.copyOf(sources);
+  }
+
+  @Override
+  public List<T> getCredentialsDefinitions() {
+    return sources.stream()
+        .flatMap(source -> source.getCredentialsDefinitions().stream())
```

Comment:
What is taking precedence if there are two accounts with the same name in different sources?

Reply:
Right now, it would be based on the ordering of bean definitions which isn't that ideal. Any suggestions on conflict strategies to apply here? Like throwing an exception or logging a warning or what?

Reply:
Part of me wants to allow only one source, but then I suspect it will be helpful in some situations to have some "bootstrap" accounts in e.g. clouddriver-local.yaml and then have more this way....and there isn't really a precedence issue unless more than one source defines an account with the same name.  And I guess it makes sense for clouddriver-local.yaml to take precedence...but with dynamic refresh also possible via spring cloud config I'm not sure that's possible/realistic/a good idea....

Does it make sense to add "if account name already exists, then log a warning, skip this account, but continue processing other accounts" logic to "all the places" ?  Or at least here?

Reply:
I think logging a warning that the account definition will be skipped as a duplicate.

Reply:
Great.  Is putting it here enough, or are there other places that are now subject to this issue?

Reply:
This is where the bean definitions would all be merged for a specific account type. It's reusable for any other credentials classes that opt in to the system.

Reply:
For context, I initially developed this to merge in existing sources because I have another custom source for Kubernetes credentials used on some clusters involving a CRD to configure these accounts. Combined with a naming scheme, I didn't expect to see any conflicts, though it's definitely a good idea to warn about them.

Reply:
Updated with logs and filtering dupes.